src/factorizations.jl

```diff
@@ -1,25 +1,42 @@
 using LinearAlgebra: LinearAlgebra
-using MatrixAlgebraKit:
-  eig_full!,
-  eig_trunc!,
-  eig_vals!,
-  eigh_full!,
-  eigh_trunc!,
-  eigh_vals!,
-  left_null!,
-  left_orth!,
-  left_polar!,
-  lq_full!,
-  lq_compact!,
-  qr_full!,
-  qr_compact!,
-  right_null!,
-  right_orth!,
-  right_polar!,
-  svd_full!,
-  svd_compact!,
-  svd_trunc!,
-  svd_vals!
+using .MatrixAlgebra: MatrixAlgebra
+using MatrixAlgebraKit: MatrixAlgebraKit
+
+function factorize_with(f, A::AbstractArray, biperm::BlockedPermutation{2}; kwargs...)
```

Comment:
By default this does not specialize on `f`, so this might alter type stability a bit (make it less likely that the constprop of the kwargs that influence the output type work)
Since we're using this in the eval loop anyways, it might make sense to simply copy-paste this there directly instead?